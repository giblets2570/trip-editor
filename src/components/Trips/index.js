import React, { Component } from 'react';

import { connect } from 'react-redux'

import moment from 'moment';

import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink
} from 'reactstrap';

import CreateTrip from '../CreateTrip'
import TripCard from '../TripCard'

import { DateRangePicker, SingleDatePicker, DayPickerRangeController } from 'react-dates';

import { logout } from '../../actions/authActions'
import { fetch } from '../../actions/tripsActions'

class Trips extends Component {
  constructor(){
    super();
    this.state = {
      toggled: false,
      modal: false
    }
    this.logout = this.logout.bind(this);
    this.toggle = this.toggle.bind(this);
    this.printMonth = this.printMonth.bind(this);
    this.toggleModal = this.toggleModal.bind(this);
  }
  componentWillMount(){
    this.props.dispatch(fetch());
  }
  handleChange(event,key) {
    this.setState({
      [key]: event.target.value
    });
  }
  printMonth() {
    const now = moment();
    const nextMonth = moment(now).add(1, 'months');
    const trips = this.props.trips.filter((trip) => {
      let date = moment(trip.startDate);
      return date.isBetween(now, nextMonth);
    });
  }
  toggleModal() {
    this.setState({
      modal: !this.state.modal
    })
  }
  toggle() {
    this.setState({
      toggled: !this.state.toggled
    })
  }
  logout() {
    this.props.dispatch(logout());
  }
  render() {
    const trips = this.props.trips
    .sort((a,b) => new Date(a.startDate) > new Date(b.startDate))
    .map((trip, key) => {
      return (
        <div key={key}>
          <TripCard trip={trip}/>
        </div>
      )
    });
    return (
      <div>
        <Navbar color="faded" light toggleable>
          <NavbarToggler right onClick={this.toggle} />
          <NavbarBrand to="#">Hi {this.props.user.name}</NavbarBrand>
          <Collapse isOpen={this.state.toggled} navbar>
            <Nav className="ml-auto" navbar>
              <NavItem>
                <NavLink href="#" onClick={this.printMonth}>Print Month</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="#" onClick={this.toggleModal}>Create Trip</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="#" onClick={this.logout}>Logout</NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
        <CreateTrip isOpen={this.state.modal} toggle={this.toggleModal}/>
        {trips}
      </div>
    )
  }
}

export default connect((store) => {
  return {
    user: store.auth.user,
    trips: store.trips.trips
  }
})(Trips);