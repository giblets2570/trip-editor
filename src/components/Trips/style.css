.tripsBody {
	padding: 40px;
}