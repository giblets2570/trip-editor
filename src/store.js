import { compose, applyMiddleware, createStore } from "redux"
import { persistStore, autoRehydrate } from "redux-persist"

import { createLogger }  from "redux-logger"
import thunk from "redux-thunk"
import promise from "redux-promise-middleware"

import reducer from "./reducers"

const middleware = applyMiddleware(promise(), thunk, createLogger())

const store = createStore(
	reducer, 
	undefined,
	compose(
		middleware,
		// autoRehydrate()
	)
)

// persistStore(store);

export default store;