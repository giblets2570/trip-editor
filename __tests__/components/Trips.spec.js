import React from 'react';
import renderer from 'react-test-renderer';
import Trips from '../../src/components/Trips';
import store from '../../src/store'
import { Provider } from 'react-redux'

describe('Trips (Snapshot)', () => {
  it('Trips renders', () => {
  	const match = {
  		params: {
  			id: '1234'
  		}
  	}
    let wrapper = shallow(
      <Trips match={match} store={mockStore} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('trips')).toEqual([]);
    expect(wrapper.prop('filters').destination).toEqual("");
    expect(wrapper.prop('filters').startDate).toEqual(null);
    expect(wrapper.prop('filters').endDate).toEqual(null);
  });
});