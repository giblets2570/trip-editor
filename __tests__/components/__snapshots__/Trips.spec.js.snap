// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trips (Snapshot) Trips renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Trips
    dispatch={[Function]}
    filters={
      Object {
        "destination": "",
        "endDate": null,
        "startDate": null,
      }
    }
    match={
      Object {
        "params": Object {
          "id": "1234",
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      }
    }
    trips={Array []}
  />,
  "nodes": Array [
    <Trips
      dispatch={[Function]}
      filters={
        Object {
          "destination": "",
          "endDate": null,
          "startDate": null,
        }
      }
      match={
        Object {
          "params": Object {
            "id": "1234",
          },
        }
      }
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
      storeSubscription={
        Subscription {
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        }
      }
      trips={Array []}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(Trips)
        match={
          Object {
            "params": Object {
              "id": "1234",
            },
          }
        }
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
          "storeSubscription": undefined,
        },
        "notifyNestedSubs": [Function],
        "props": Object {
          "match": Object {
            "params": Object {
              "id": "1234",
            },
          },
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "propsMode": true,
        "refs": Object {},
        "renderCount": 0,
        "selector": Object {
          "error": null,
          "props": Object {
            "dispatch": [Function],
            "filters": Object {
              "destination": "",
              "endDate": null,
              "startDate": null,
            },
            "match": Object {
              "params": Object {
                "id": "1234",
              },
            },
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "trips": Array [],
          },
          "run": [Function],
          "shouldComponentUpdate": false,
        },
        "setWrappedInstance": [Function],
        "state": Object {},
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 5,
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Trips
          dispatch={[Function]}
          filters={
            Object {
              "destination": "",
              "endDate": null,
              "startDate": null,
            }
          }
          match={
            Object {
              "params": Object {
                "id": "1234",
              },
            }
          }
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
          storeSubscription={
            Subscription {
              "listeners": Object {
                "notify": [Function],
              },
              "onStateChange": [Function],
              "parentSub": undefined,
              "store": Object {
                "clearActions": [Function],
                "dispatch": [Function],
                "getActions": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
              },
              "unsubscribe": null,
            }
          }
          trips={Array []}
        />,
        "_debugID": 2,
        "_renderedOutput": <Trips
          dispatch={[Function]}
          filters={
            Object {
              "destination": "",
              "endDate": null,
              "startDate": null,
            }
          }
          match={
            Object {
              "params": Object {
                "id": "1234",
              },
            }
          }
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
          storeSubscription={
            Subscription {
              "listeners": Object {
                "notify": [Function],
              },
              "onStateChange": [Function],
              "parentSub": undefined,
              "store": Object {
                "clearActions": [Function],
                "dispatch": [Function],
                "getActions": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
              },
              "unsubscribe": null,
            }
          }
          trips={Array []}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(Trips)
    match={
      Object {
        "params": Object {
          "id": "1234",
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
}
`;
